hypothesis-python/src/hypothesis/internal/floats.py

```diff
@@ -11,22 +11,58 @@
 import math
 import struct
 from sys import float_info
-from typing import Callable, Optional, SupportsFloat
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Dict,
+    Literal,
+    Optional,
+    SupportsFloat,
+    Tuple,
+    Union,
+    overload,
+)
+
+if TYPE_CHECKING:
+    from typing import TypeAlias
+else:
+    TypeAlias = object
+
+SignedIntFormat: "TypeAlias" = Literal["!h", "!i", "!q"]
+UnsignedIntFormat: "TypeAlias" = Literal["!H", "!I", "!Q"]
+IntFormat: "TypeAlias" = Union[SignedIntFormat, UnsignedIntFormat]
+FloatFormat: "TypeAlias" = Literal["!e", "!f", "!d"]
 
 # Format codes for (int, float) sized types, used for byte-wise casts.
 # See https://docs.python.org/3/library/struct.html#format-characters
-STRUCT_FORMATS = {
+STRUCT_FORMATS: Dict[int, Tuple[UnsignedIntFormat, FloatFormat]] = {
     16: ("!H", "!e"),
     32: ("!I", "!f"),
     64: ("!Q", "!d"),
 }
 
+TO_SIGNED_FORMAT: Dict[UnsignedIntFormat, SignedIntFormat] = {
+    "!H": "!h",
+    "!I": "!i",
+    "!Q": "!q",
+}
+
+
+@overload
+def reinterpret_bits(x: float, from_: str, to: IntFormat) -> int: ...
+
+
+@overload
+def reinterpret_bits(x: float, from_: str, to: FloatFormat) -> float: ...
```

Comment:
Honestly I'd be comfortable just dropping an `Any` or a `# type: ignore` here - let's keep the code as simple as possible for now, and worry about completeness if or when someone e.g. tries to compile it.

Reply:
Fair! Will keep this 95% solution ideal in mind going forward.